source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -53,16 +53,29 @@ Http::FilterHeadersStatus ProxyFilter::decodeHeaders(Http::RequestHeaderMap& hea
   }
   cluster_info_ = cluster->info();
 
-  auto& resource = cluster_info_->resourceManager(route_entry->priority()).pendingRequests();
```

Comment:
This is a breaking change which makes me uncomfortable. I think we need a runtime feature flag here to preserve the old behavior for a period of time, as well as documentation around it.

Reply:
The dynamic forward proxy is marked alpha (`envoy_cc_extension(status="alpha")`) and also clearly documented in https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/http/http_proxy as well, so I don't feel strongly to introduce a runtime feature flag to guard this. Are we aware of anyone is relying on this behavior?

Reply:
I know people use this in production already, so I think we need to remove the alpha tag (and the protos are not alpha in v3). With that said, given that we have alpha now, I'm willing to let this slide, but we need to have a very specific release note about this behavior change. Please do a follow up to remove the alpha tag and other pre-prod verbiage.

Reply:
Introduced the runtime flag. This change is needed to change documents. https://github.com/envoyproxy/envoy/pull/11028/files#diff-e3f6d903dd6231c359a06a057d7bdc7bR62